apps/document-check-function/check-new-document/index.js

```diff
@@ -0,0 +1,87 @@
+import got, { HTTPError } from 'got';
+import { isEqual } from 'lodash-es';
+import { sendDocumentStateAction } from './back-office-api-client.js';
+import { scanStream } from './scan-stream.js';
+
+/**
+ * @typedef {{bindingData: {uri: string}, error: any, info: any}} Context
+ */
+
+/**
+ * @param {boolean} isInfected
+ * @returns {string}
+ */
+const mapIsInfectedToMachineAction = (isInfected) => {
+	return isInfected ? 'check_fail' : 'check_success';
+};
+
+/**
+ * TODO: At the minute, we get the blob case ID and GUID from the path.
+ * Once we save the same info to the Blob Metadata, let's get it from there
+ *
+ * @param {string} blobUri
+ * @returns {{caseId: string, guid: string}}
+ */
+const getBlobCaseIdAndGuid = (blobUri) => {
+	const uriParts = blobUri.split('/');
+
+	if (uriParts.length !== 5 || uriParts[1] !== 'applications') {
+		throw new Error(`Unexpected blob URI ${blobUri}`);
```

Comment:
For expected errors like this (and for example, where you throw an error if the state change has already been completed), are the errors being propagated to the top-level? We want to make sure that the functions only 'throw' errors for things that we can potentially retry, because we want to be able to retry for transient errors (e.g. if clamav container is down) but not retry for expected errors.